src/DeploymentUtils.ts

```diff
@@ -1,19 +1,47 @@
-import * as deployments from "../deployments/deployments.json";
+import * as deployments_ from "../deployments/deployments.json";
+interface DeploymentExport {
+  [chainId: string]: { [contractName: string]: { address: string; blockNumber: number } };
+}
+const deployments: DeploymentExport = deployments_ as any;
 
 // Returns the deployed address of any contract on any network.
 export function getDeployedAddress(contractName: string, networkId: number): string {
   try {
-    return (deployments as any)[networkId.toString()][contractName].address;
+    return deployments[networkId.toString()][contractName].address; // First try match directly on the name provided.
   } catch (_) {
+    try {
+      // If this does not match then try search on a Regex on SpokePool to let the caller exclude the chain name.
+      for (const _contractName of Object.keys(deployments[networkId.toString()]))
+        if (/.*_SpokePool/.test(_contractName)) return deployments[networkId.toString()][_contractName].address;
```

Comment:
this could error because of a typo, and then the user ends up getting spoke pool address when maybe they meant somethign completely different, which could be bad. 

I'd suggest doing something where you are finding in an array of keys if they include "contractName".  Since this logic is duplicated, id suggest making it a function.

somethign like:
```
function findContractName(partial){
  const found = [...Object.keys(deployments[networkId])].find(name=>name.includes(partial))
  if(found) return deployments[networkId.toString()][found]
  throw new Error('Unable to find contract matching ' + partial)
}
```

Reply:
the only way that would happen is if you added `SpokePool` to the search. this regex will ONLY pass if `*_SpokePool` is set. you are right though that you could do something silly like `HubPool_SpokePool` which would return the spoke pool on that chain.

Reply:
however, I have modified this to export the name of the contract as `SpokePool` on each respective chain to avoid needing this workaround alltogether.

Reply:
I see, i missed that check